### PR TITLE
Add notes on collecting coverage data as artifacts

### PR DESCRIPTION
**Pull request by teesloane**
We used to have a document on collecting code-coverage in docs 1.0. The `collecting test data` document seems to cover a lot of the same content (and adds more), but with regard to newer features (ie, the in-app test summary view). 

This PR add a small section to let user's know that they can also generate and upload code-coverage results as an artifact if they don't want to use junit xml reporting.

### CODE REVIEW

File: jekyll/_cci2/collect-test-data.md [hunk 1]
```diff
@@ -476,6 +476,29 @@ For more details on `--runInBand`, refer to the [Jest CLI](https://facebook.gith
 
 To access test metadata for a run from the API, refer to the [test-metadata API documentation]( {{ site.baseurl }}/api/v1-reference/#test-metadata).
 
+## Storing Test Coverage Reports
+
+If you would like to upload code coverage reports, or other non-junit
+test-results, consider using a `store-artifacts` step. In the example below,
```
Comment (michelle-luna) [inline jekyll/_cci2/collect-test-data.md]: remove hyphen in test results

File: jekyll/_cci2/collect-test-data.md [hunk 2]
```diff
@@ -476,6 +476,29 @@ For more details on `--runInBand`, refer to the [Jest CLI](https://facebook.gith
...
+Jest is used with the `--collectCoverage` flag enabled, which can then upload
```
Comment (michelle-luna) [inline jekyll/_cci2/collect-test-data.md]: Should this be part of the Jest section above? instead of a section ## below it?

**Comment by teesloane** [comment]
After thinking about it, there's lots more documentation I can put in for code coverage— especially more language examples and using third party providers. I'm going to close this for now, add more details, and re-open.

**Comment by teesloane** [comment]
Reopening this:
- migrate code coverage doc from 1 -> 2.
- Add examples for ruby, java, go, javascript, ands php.

**Comment by michelle-luna** [comment]
Cool, but the title in the sidebar should match the title of the doc, so when a user clicks the sidebar, they see that same title on the doc that loads (either title is fine with me, just make them the same). Thank you!

**Pull request merged**